section4/reflection.md

```diff
@@ -1,21 +1,37 @@
 
 1. What is an Object, and how is it different from an Array in Javascript?
+- An Object is a variable that contains properties as key-value pairs and methods. All of the properties are related to the object. For example 'var nina' would contain different attributes related to me - my height, age, weight, hairColor, eyeColor, etc. Arrays are list-like objects. They contain elements that can be accessed by index number. They typically contain similar elements. For example  'var fruits' would contain many different fruits. But arrays could contain different attributes of a particular fruit. They create and store lists of data. You can also store objects in an array. And you can store arrays in objects.
 
 1. For each set of data, would an array or object be better to store it? Explain your choice.
 
-  * List of all of the students in class
-  * List of states and their capitals
-  * List of things to pack for vacation
-  * Names of all the Instagram accounts I follow
-  * List of student names and their cohort
-  * Ingredients and amount of each ingredient to bake a cake
-  * All my favorite restaurants
+  * List of all of the students in class - Array
+  * List of states and their capitals - Object
+  * List of things to pack for vacation - Array
+  * Names of all the Instagram accounts I follow - Array
+  * List of student names and their cohort - Object
+  * Ingredients and amount of each ingredient to bake a cake - Object
+  * All my favorite restaurants - Array
 
-1. In this section, we talked about an `transaction item object`. It has keys that represent different properties an bank transaction would have, and values that store the data. In a banking app, this data would eventually be displayed in the browser for the user. Think of another Object in a web application that you use frequently. What is it? What keys might be on it? What are example values? 
+1. In this section, we talked about an `transaction item object`. It has keys that represent different properties a bank transaction would have, and values that store the data. In a banking app, this data would eventually be displayed in the browser for the user. Think of another Object in a web application that you use frequently. What is it? What keys might be on it? What are example values?
+```
+class AmazonOrder {
+  constructor (item, amt, purchaseDate, deliveryDate, supplier) {
+    this.item = item;   //(what the person purchased)
+    this.amount = amt;  //(what the cost was)
+    this.purchaseDate = purchaseDate;
+    this.deliveryDate = deliveryDate;
+    this.supplier = supplier;
+    this.deliveryStatus = deliveryStatus;
+  }
+}
+  ```
 
 1. What questions do you still have about classes and/or Objects?
+- Can you only use a method on the key, not the value?
+- Also, if there are anymore code challenges like the 'class Tweet' challenge, I would love to get more practice. Integrating the methods into the classes is more challenging than I anticipated.
```

Comment:
Look for "beginner javascript challenges" or check out the entire Javascript track on exercism.io!